journal/src/main/scala/com/evolutiongaming/kafka/journal/HeadCacheConsumption.scala

```diff
@@ -22,11 +20,11 @@ object HeadCacheConsumption {
   def apply[F[_] : BracketThrowable : Sleep](
     topic: Topic,
     pointers: F[Map[Partition, Offset]],
-    consumer: Resource[F, TopicCache.Consumer[F]],
+    consumer: TopicCache.Consumer[F],
```

Comment:
this will exclude creation of consume from `retry` scope… we should re-create consume during retry

Reply:
yes, I've realised it after test failed. Now reworking this approach to left same behaviour as it was

Reply:
I've reverted the logic back & Kafka consumer recreates in `retry` scope again. Additionally instance of consumer periodically created to update in-memory partition cache, please find usage of `ExpiringRef` that incapsulates it